Source/DafnyCore/Compilers/Compiler-java.cs

```diff
@@ -1006,9 +1022,14 @@ protected class ClassWriter : IClassWriter {
       } else if (e.Value is bool value) {
         wr.Write(value ? "true" : "false");
       } else if (e is CharLiteralExpr) {
-        wr.Write($"'{(string)e.Value}'");
+        var v = (string)e.Value;
+        if (UnicodeChars && Util.MightContainNonAsciiCharacters(v, false)) {
+          wr.Write($"{Util.UnescapedCharacters(v, false).Single()}");
```

Comment:
Even it it contains non-ascii, can't we still use `'…'` in Go?

Reply:
This is the Java compiler :)